### PR TITLE
Feature dropping functionality

### PR DESCRIPTION
**Pull request by arahayrabedian**
- So find how to drop certain feature columns from the cleaned data before doing any processing, it's a little crude, using the csv with the feature lengths, but oh well it seems to work.
-  takes the opportunity to pop in n_jobs on the RandomForestRegressor.
- bumps tensorflow version, the old one seems to have evaporated from pypi

I've built my own 'main' for this which i'm not including here, it will run RandomForestRegressor while dropping the columns: `['contract_type', 'contract_time', 'company', 'source', 'town', 'region', 'job_titles', 'categories']` in all their combinations, ~255 - currently running.

Appreciate if someone can point out bugs, i'm sure they exist.

### CODE REVIEW

File: src/models/BaseModel.py [hunk 1]
```diff
@@ -90,6 +106,35 @@ def __init__(self, train_size=0.75, test_size=None, force_load_all=False):
         print('Train test split complete \n')
         self.mae_test_error = -1
 
+    def _get_feature_indices(self, feature_name):
+        range_start = 0
```
Comment (arahayrabedian) [inline src/models/BaseModel.py]: note to self: add comments here

File: requirements.txt [hunk 2]
```diff
@@ -9,7 +9,6 @@ python-rake>=1.0.7
 sklearn
 scipy==0.18.0
 keras==1.2.1
-tensorflow==0.12.1
+tensorflow==1.1.0
```
Comment (blanche) [inline requirements.txt:12]: not sure if we want that - all of our previous stuff relies on 0.12 ?

File: src/models/BaseModel.py [hunk 3]
```diff
@@ -197,9 +242,12 @@ def load_cleaned_data():
         loaded_data = pd.read_csv(BaseModel.CLEANED_FILE_NAME, header=None)
         loaded_data = loaded_data.as_matrix()
         loaded_data = loaded_data.astype(int)
-        return loaded_data
 
-    abc.abstractmethod
+        loaded_feature_data = pd.read_csv(BaseModel.FEATURE_METADATA_FILE_NAME,
```
Comment (blanche) [inline src/models/BaseModel.py:254]: where is this file written?

File: src/models/BaseModel.py [hunk 4]
```diff
@@ -75,12 +88,15 @@ def __init__(self, train_size=0.75, test_size=None, force_load_all=False):
...
-            test_size=test_size,
-            random_state=1)
+        try:
+            self.description_train_data, self.description_test_data = train_test_split(
+                self.description_feature,
+                train_size=train_size,
+                test_size=test_size,
+                random_state=1)
+        except:
+            print("Not loading description training/test data because "
```
Comment (blanche) [inline src/models/BaseModel.py]: when is that the case?

File: requirements.txt (hunk 2 above)
Comment (arahayrabedian) [reply requirements.txt:12]: `0.12.1` is no longer on pypi. works fine on my local with `1.1.0` (at least the parts i'm running) - i'd already say that's lesser of two breaks.

File: src/models/BaseModel.py (hunk 4 above)
Comment (arahayrabedian) [reply src/models/BaseModel.py]: https://github.com/soton-data-mining/job-salary-prediction/blob/1b75070a0ff83927312437bfc36119832eba55d3/src/models/BaseModel.py#L45-L53

when `force_load_all` is False

File: src/models/BaseModel.py (hunk 3 above)
Comment (arahayrabedian) [reply src/models/BaseModel.py:254]: by filename, kept it this way for consistency with how other files are getting written/read

File: src/models/BaseModel.py [hunk 5]
```diff
@@ -75,12 +88,15 @@ def __init__(self, train_size=0.75, test_size=None, force_load_all=False):
...
-            train_size=train_size,
-            test_size=test_size,
-            random_state=1)
+        try:
+            self.description_train_data, self.description_test_data = train_test_split(
+                self.description_feature,
+                train_size=train_size,
+                test_size=test_size,
+                random_state=1)
+        except:
```
Comment (arahayrabedian) [inline src/models/BaseModel.py]: note to self: add exact exception

Comment (blanche) [reply src/models/BaseModel.py]: i would rather check that with an `if job_description exists` but that's a matter of taste

Comment (arahayrabedian) [reply src/models/BaseModel.py]: yeah for me it depends, in this case i don't think it's an intentional flow and i'd rather handle it with a try/except. If it was an intentional, well defined, flow i would have taken an if

File: requirements.txt (hunk 2 above)
Comment (alexdy2007) [reply requirements.txt:12]: I tested it. works absolutly fine with 1.1.0 as needed 1.1.0 to run keras with gpu

**Review by blanche — APPROVED** [review]
"good job"

**Pull request merged**